src/core/load-fragment.c

```diff
@@ -5423,6 +5423,87 @@ int config_parse_ip_filter_bpf_progs(
         return 0;
 }
 
+int config_parse_restrict_network_interfaces(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+        CGroupContext *c = data;
+        bool is_allow_rule = true;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        if (isempty(rvalue)) {
+                /* Empty assignment resets the list */
+                c->restrict_network_interfaces = set_free_free(c->restrict_network_interfaces);
+                return 0;
+        }
+
+        if (rvalue[0] == '~') {
+                is_allow_rule = false;
+                rvalue++;
+        }
+
+        if (!c->restrict_network_interfaces) {
+                c->restrict_network_interfaces = set_new(NULL);
+                if (!c->restrict_network_interfaces)
+                        return log_oom();
+
+                /* Only initialize this when creating the set */
+                c->restrict_network_interfaces_is_allow_list = is_allow_rule;
+        }
+
+        for (const char *p = rvalue;;) {
+                _cleanup_free_ char *word = NULL;
+                char *it;
+
+next:
+                r = extract_first_word(&p, &word, NULL, EXTRACT_UNQUOTE);
+                if (r == 0)
+                        break;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0) {
+                        log_syntax(unit, LOG_WARNING, filename, line, r,
+                                   "Trailing garbage in %s, ignoring: %s", lvalue, rvalue);
+                        break;
+                }
```

Comment:
should validate the ifname with ifname_valid() here and refuse invalid ifnames

Reply:
I'm validating the names while installing it, isn't that enough? If not, what do you mean by refuse, ignore them or return an error indicating the ifname is not valid?

Reply:
we generally want to refuse stuff early on, while parsing config.

i.e. i am not suggesting to resolve the names, i.e. checking whether they exist. That should only happen when applying things. All I am saying is to use ifname_valid() (or ifname_valid_full()?) here to check if the specified name superficially looks like something that could be an ifname, and if not refuse with a log_syntax() warning right-away.

Reply:
I'm not sure I follow this idea. In the current implementation we're **not** refusing to load the unit when there is a non existing interface. There is just a warning message but the loading continues... Do you think we should change it and fail if there is an interface that can't be resolved to its ifindex? 

https://github.com/systemd/systemd/blob/c70b704088c657495f0628373c3d96331156a9be/src/core/restrict-ifaces.c#L58

Reply:
ifname_valid() doesn't check for existance, but simply checks that the string is suitable as ifname, i.e. that it isn#t overly long, or an empty string or has weird chars and a bunch of other things. we should do such superficial validation for everything we parse.

Reply:
Ah, got it. Thanks for the clarification.